### ISSUE TITLE
The installer package includes files that are only readable by the root user (3.13.0)

### ISSUE THREAD

**Issue by adil192** [flutter/flutter#132725]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Follow https://docs.flutter.dev/deployment/macos#create-a-build-archive-with-codemagic-cli-tools up to and including `app-store-connect publish --path "$PACKAGE_NAME"`

### Expected results

Publishing to the App Store works

### Actual results

```
2023-08-17 01:59:37.169  INFO: Show Progress: Waiting for App Store Connect SPI analysis response…
2023-08-17 02:01:23.077  INFO: Show Progress: Upload failed.
Asset validation failed
The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app. (ID: 7e9d5e25-9883-4df3-a89f-b2816e3e1723)
2023-08-17 02:01:23.077  INFO: 
=============
UPLOAD FAILED with 1 error.
=============
{"tool-version":"6.043.14043","tool-path":"\/Applications\/Xcode_14.2.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"12.6.7","product-errors":[{"message":"Asset validation failed","userInfo":{"NSLocalizedFailureReason":"The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app. (ID: 7e9d5e25-9883-4df3-a89f-b2816e3e1723)","NSLocalizedDescription":"Asset validation failed","original_server_error":{"status":"409","id":"7e9d5e25-9883-4df3-a89f-b2816e3e1723","code":"STATE_ERROR.VALIDATION_ERROR.90255","title":"Asset validation failed","detail":"The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app."}},"code":90255}]}


Failed to upload archive at "build/macos/Build/Products/Release/Ricochlime.pkg"
Failed to publish build/macos/Build/Products/Release/Ricochlime.pkg
Error: Process completed with exit code 1.
```

### Code sample

I haven't been able to reproduce this since to get to the point of publishing to the App Store, you need to set up all of the certificates etc. I'm guessing this occurs in a blank Flutter project though.

I can confirm, however, that reverting to Flutter 3.10.6 doesn't have this issue.

### Logs

<details><summary>Logs</summary>

```console
Publish "build/macos/Build/Products/Release/Ricochlime.pkg" to App Store Connect
Bundle identifier: com.adilhanney.ricochlime
Install size: 51919KB
Version code: 5000
Version: 0.5.0

Upload "build/macos/Build/Products/Release/Ricochlime.pkg" to App Store Connect
2023-08-17 01:58:52.863  INFO: ContentDelivery version 2.141.101 (141101).
2023-08-17 01:58:52.866  INFO: Show Progress: Contacting Apple Services…
2023-08-17 01:58:53.253  INFO: Received build ID: 2ad9c639-c1cd-428d-8852-fda1f2482567
2023-08-17 01:58:53.254  INFO: Show Progress: Making copy of ‘Ricochlime.pkg’…
2023-08-17 01:58:53.258  INFO: Show Progress: Preparing to upload ‘Ricochlime.pkg’…
2023-08-17 01:58:53.323  INFO: Show Progress: Analyzing package…
2023-08-17 01:59:27.503  INFO: Show Progress: Sending analysis to App Store Connect…
2023-08-17 01:59:27.658  INFO: COMPLETED - PART 1 - asset-description-7A7A7DB4-CA1D-4C43-8C0C-6C16908C3832.xml - eTag: "13390147FC1CACB9B3A15C2BAFE1905F"
2023-08-17 01:59:27.662  INFO: Show Progress: Waiting for App Store Connect analysis response…
2023-08-17 01:59:36.215 *** Error: ERROR: Asset validation failed (90255) The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app. (ID: 7e9d5e25-9883-4df3-a89f-b2816e3e1723)
2023-08-17 01:59:36.663  INFO: Show Progress: Sending SPI analysis to App Store Connect…
2023-08-17 01:59:36.994  INFO: COMPLETED - PART 1 - DTAppAnalyzerExtractorOutput-80C2D7FD-2FBA-4C4C-933E-D64EBF5B72D2.zip - eTag: "A3D0EA536B66AEADC014CD3643C3C615"
2023-08-17 01:59:37.077  INFO: COMPLETED - PART 4 - DTAppAnalyzerExtractorOutput-80C2D7FD-2FBA-4C4C-933E-D64EBF5B72D2.zip - eTag: "9E53CF42A803A4BC10D6521FD748E71B"
2023-08-17 01:59:37.092  INFO: COMPLETED - PART 2 - DTAppAnalyzerExtractorOutput-80C2D7FD-2FBA-4C4C-933E-D64EBF5B72D2.zip - eTag: "11355DC834E9AF31B82E04048AA496B1"
2023-08-17 01:59:37.165  INFO: COMPLETED - PART 3 - DTAppAnalyzerExtractorOutput-80C2D7FD-2FBA-4C4C-933E-D64EBF5B72D2.zip - eTag: "F1A76EEED562D78C1171AAC230695A1B"
2023-08-17 01:59:37.169  INFO: Show Progress: Waiting for App Store Connect SPI analysis response…
2023-08-17 02:01:23.077  INFO: Show Progress: Upload failed.
Asset validation failed
The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app. (ID: 7e9d5e25-9883-4df3-a89f-b2816e3e1723)
2023-08-17 02:01:23.077  INFO: 
=============
UPLOAD FAILED with 1 error.
=============
{"tool-version":"6.043.14043","tool-path":"\/Applications\/Xcode_14.2.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"12.6.7","product-errors":[{"message":"Asset validation failed","userInfo":{"NSLocalizedFailureReason":"The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app. (ID: 7e9d5e25-9883-4df3-a89f-b2816e3e1723)","NSLocalizedDescription":"Asset validation failed","original_server_error":{"status":"409","id":"7e9d5e25-9883-4df3-a89f-b2816e3e1723","code":"STATE_ERROR.VALIDATION_ERROR.90255","title":"Asset validation failed","detail":"The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app."}},"code":90255}]}


Failed to upload archive at "build/macos/Build/Products/Release/Ricochlime.pkg"
Failed to publish build/macos/Build/Products/Release/Ricochlime.pkg
Error: Process completed with exit code 1.
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

Note that this was run on the GitHub Actions runner:

```console
  ╔════════════════════════════════════════════════════════════════════════════╗
  ║                 Welcome to Flutter! - https://flutter.dev/                  ║
  ║                                                                            ║
  ║ The Flutter tool uses Google Analytics to anonymously report feature usage ║
  ║ statistics and basic crash reports. This data is used to help improve      ║
  ║ Flutter tools over time.                                                   ║
  ║                                                                            ║
  ║ Flutter tool analytics are not sent on the very first run. To disable      ║
  ║ reporting, type 'flutter config --no-analytics'. To display the current    ║
  ║ setting, type 'flutter config'. If you opt out of analytics, an opt-out    ║
  ║ event will be sent, and then no further information will be sent by the    ║
  ║ Flutter tool.                                                              ║
  ║                                                                            ║
  ║ By downloading the Flutter SDK, you agree to the Google Terms of Service.  ║
  ║ Note: The Google Privacy Policy describes how data is handled in this      ║
  ║ service.                                                                   ║
  ║                                                                            ║
  ║ Moreover, Flutter includes the Dart SDK, which may send usage metrics and  ║
  ║ crash reports to Google.                                                   ║
  ║                                                                            ║
  ║ Read about data we send with crash reports:                                ║
  ║ https://flutter.dev/docs/reference/crash-reporting                         ║
  ║                                                                            ║
  ║ See Google's privacy policy:                                               ║
  ║ https://policies.google.com/privacy                                        ║
  ╚════════════════════════════════════════════════════════════════════════════╝
Could not parse java version from: 
If there is a version please look for an existing bug https://github.com/flutter/flutter/issues/ and if one does not exist file a new issue.
[✓] Flutter (Channel stable, 3.13.0, on macOS 12.6.7 21G651 darwin-x64, locale en)
    • Flutter version 3.13.0 on channel stable at /Users/runner/hostedtoolcache/flutter/stable-3.13.0-x64
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision efbf63d9c6 (25 hours ago), 2023-08-15 21:05:06 -0500
    • Engine revision 1ac611c64e
    • Dart version 3.1.0
    • DevTools version 2.25.0
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/runner/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/runner/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/runner/Library/Android/sdk
    • Java binary at: /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.382-5/x64/Contents/Home/bin/java
    ✗ Could not determine java version
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode_14.2.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.12.1
[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.6.7 21G651 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 115.0.5790.170
[✓] Network resources
    • All expected network resources are available.
! Doctor found issues in 2 categories.
The Flutter CLI developer tool uses Google Analytics to report usage and diagnostic data
along with package dependencies, and crash reporting to send basic crash reports.
This data is used to help improve the Dart platform, Flutter framework, and related tools.
Telemetry is not sent on the very first run.
To disable reporting of telemetry, run this terminal command:
flutter --disable-telemetry.
If you opt out of telemetry, an opt-out event will be sent,
and then no further information will be sent.
This data is collected in accordance with the
Google Privacy Policy (https://policies.google.com/privacy).
```

</details>

Edit: This is also happening with Saber (another one of my apps).

**Comment by huycozy** [comment]
Hi @adil192 
Which type of user did you log in when you build the app? (root or normal user?)

Regarding the error message above, can you try [this solution](https://github.com/xamarin/xamarin-macios/issues/7314#issuecomment-552079414) to see if it works?

**Comment by adil192** [comment]
> Hi @adil192 Which type of user did you log in when you build the app? (root or normal user?)

This is on a GitHub Actions runner which uses the `runner` user (which isn't root).

> Regarding the error message above, can you try [this solution](https://github.com/xamarin/xamarin-macios/issues/7314#issuecomment-552079414) to see if it works?

I tried running `xattr -rc *` on my local machine and `git status` doesn't show that anything changed. I also ran `ls -l@` to see if anything had extended attributes and it doesn't show anything:
<details>
<summary>ls -l@</summary>

```console
Adils-MacBook-Air:ricochlime adilhanney$ ls -l@ assets/images/
total 120
-rw-r--r--  1 adilhanney  staff    585 17 Aug 03:44 LICENSE.txt
-rw-r--r--  1 adilhanney  staff   5787 17 Aug 03:44 plains.png
-rw-r--r--  1 adilhanney  staff  12465 17 Aug 03:44 player.png
-rw-r--r--  1 adilhanney  staff   4413 17 Aug 03:44 slime.png
-rw-r--r--  1 adilhanney  staff   1397 17 Aug 03:44 water1.png
-rw-r--r--  1 adilhanney  staff   1410 17 Aug 03:44 water2.png
-rw-r--r--  1 adilhanney  staff   1425 17 Aug 03:44 water3.png
-rw-r--r--  1 adilhanney  staff   1427 17 Aug 03:44 water4.png
-rw-r--r--  1 adilhanney  staff   1410 17 Aug 03:44 water5.png
-rw-r--r--  1 adilhanney  staff   1430 17 Aug 03:44 water6.png

Adils-MacBook-Air:ricochlime adilhanney$ ls -l@
total 168
-rw-r--r--   1 adilhanney  staff  34260 17 Aug 03:44 LICENSE.md
-rw-r--r--   1 adilhanney  staff   1391 17 Aug 03:44 README.md
-rw-r--r--   1 adilhanney  staff   2357 17 Aug 03:44 analysis_options.yaml
drwxr-xr-x  14 adilhanney  staff    448 17 Aug 03:44 android
drwxr-xr-x   4 adilhanney  staff    128 17 Aug 03:44 appimage
drwxr-xr-x   4 adilhanney  staff    128 17 Aug 03:44 assets
drwxr-xr-x   8 adilhanney  staff    256 17 Aug 03:44 assets_raw
drwxr-xr-x   5 adilhanney  staff    160 17 Aug 03:44 flatpak
-rw-r--r--   1 adilhanney  staff    842 17 Aug 03:42 flutter_test_macos_build.iml
drwxr-xr-x   3 adilhanney  staff     96 17 Aug 03:44 installers
drwxr-xr-x   9 adilhanney  staff    288 17 Aug 03:44 ios
drwxr-xr-x   8 adilhanney  staff    256 17 Aug 03:44 lib
drwxr-xr-x   8 adilhanney  staff    256 17 Aug 03:44 linux
drwxr-xr-x  10 adilhanney  staff    320 17 Aug 03:44 macos
drwxr-xr-x   3 adilhanney  staff     96 17 Aug 03:44 metadata
drwxr-xr-x   3 adilhanney  staff     96 17 Aug 03:44 patches
-rw-r--r--   1 adilhanney  staff   1514 17 Aug 03:44 privacy_policy.md
-rw-r--r--   1 adilhanney  staff  18653 17 Aug 03:57 pubspec.lock
-rw-r--r--   1 adilhanney  staff   4333 17 Aug 03:44 pubspec.yaml
drwxr-xr-x   3 adilhanney  staff     96 17 Aug 03:44 scripts
-rw-r--r--   1 adilhanney  staff    147 17 Aug 03:44 slang.yaml
drwxr-xr-x   3 adilhanney  staff     96 17 Aug 03:44 snap
drwxr-xr-x   4 adilhanney  staff    128 17 Aug 03:44 test
drwxr-xr-x   7 adilhanney  staff    224 17 Aug 03:44 web
drwxr-xr-x   6 adilhanney  staff    192 17 Aug 03:44 windows
```
</details>

Another thing I tried (unsuccessfully) is running `chown` on the GitHub Actions workflow on everything right after cloning, after `flutter build macos`, and after packaging and signing.

**Comment by AndreiMisiukevich** [comment]
I've run into the same issue today with code magic CI.

Xcode 14.3.1
cocoapods (1.12.1)
 flutter_macos_arm64_3.13.0-stable.zip


<img width="1215" alt="image" src="https://github.com/flutter/flutter/assets/10124814/74292bac-96eb-4309-89a6-d79163c1ba31">

**Comment by huycozy** [comment]
>I tried running xattr -rc * on my local machine

@adil192 Did you run it on the project root directory? Can you try `xattr` on `*.app` after building Flutter macOS? (right after step 12th on the instruction above)

@AndreiMisiukevich Please try the same.

**Comment by julienbm** [comment]
Same problem here:

```
~ xattr MyApp.app  :                                                                                                                                            
com.apple.provenance
```

**Comment by JulienDev** [comment]
Same problem here:

```
~ xattr MyApp.app  :
com.apple.provenance
```

**Comment by JulienDev** [comment]
As a workaround I had to downgrade to 3.10.6 :(
`flutter downgrade v3.10.6`

**Comment by jayjah** [comment]
> > I tried running xattr -rc * on my local machine
> 
> @adil192 Did you run it on the project root directory? Can you try `xattr` on `*.app` after building Flutter macOS? (right after step 12th on the instruction above)
> 
> @AndreiMisiukevich Please try the same.
> 
> UPDATE: If it can not make this work, please try the suggested solution here: https://github.com/orgs/codemagic-ci-cd/discussions/1498#discussioncomment-5381028

Same problem here - suggested workaround in `codemagic-ci-cd/discussions` does not work for me.

**Comment by adil192** [comment]
> > I tried running xattr -rc * on my local machine
> 
> @adil192 Did you run it on the project root directory? Can you try `xattr` on `*.app` after building Flutter macOS? (right after step 12th on the instruction above)
> 
> @AndreiMisiukevich Please try the same.
> 
> UPDATE: If it can not make this work, please try the suggested solution here: https://github.com/orgs/codemagic-ci-cd/discussions/1498#discussioncomment-5381028

Here are the commands I ran (on the GitHub Actions runner):
```yaml
  - name: Build for macOS
    run: |
      flutter build macos
      ls -l@ build/macos/Build/Products/Release
      xattr build/macos/Build/Products/Release/*.app
      echo 'Removing extended attributes'
      xattr -rc build/macos/Build/Products/Release/*.app
      echo 'Setting frag shader permissions'
      find build/macos/Build/Products/Release -name "*.frag" -exec chmod 0644 {} \;
```
and the output:
```console
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Running pod install...                                           1,758ms
Building macOS application...                                   
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
total 0
drwxr-xr-x  5 runner  staff  160 Aug 21 15:57 App.framework
drwxr-xr-x  3 runner  staff   96 Aug 21 15:57 App.framework.dSYM
drwxr-xr-x@ 2 runner  staff   64 Aug 21 15:54 FlutterMacOS
	com.apple.xcode.CreatedByBuildSystem	  4 
drwxr-xr-x  7 runner  staff  224 Aug 16 17:15 FlutterMacOS.framework
drwxr-xr-x  7 runner  staff  224 Aug 21 15:54 Pods_Runner.framework
drwxr-xr-x  3 runner  staff   96 Aug 21 15:57 Ricochlime.app
drwxr-xr-x  3 runner  staff   96 Aug 21 15:57 Ricochlime.app.dSYM
drwxr-xr-x  9 runner  staff  288 Aug 21 15:57 Ricochlime.swiftmodule
drwxr-xr-x@ 4 runner  staff  128 Aug 21 15:54 path_provider_foundation
	com.apple.xcode.CreatedByBuildSystem	  4 
drwxr-xr-x@ 4 runner  staff  128 Aug 21 15:54 shared_preferences_foundation
	com.apple.xcode.CreatedByBuildSystem	  4 
Removing extended attributes
Setting frag shader permissions
```

Publishing to the App Store fails with the same error. I've attached the full logs if they're relevant.
[logs_409.zip](https://github.com/flutter/flutter/files/12398355/logs_409.zip)

**Comment by adil192** [comment]
@huycozy I wasn't able to publish via Xcode since selecting [Product > Archive](https://docs.flutter.dev/deployment/macos#create-a-build-archive-with-xcode) always yields "Build failed".

However, I extracted the package file with `pkgutil --expand Ricochlime.pkg Ricochlimepkg` and ran `ls -al@ Ricochlimepkg/`:
```console
Adils-MacBook-Air:Downloads adilhanney$ ls -al@ Ricochlimepkg/
total 8
drwxr-xr-x   4 adilhanney  staff   128 23 Aug 14:22 .
drwx------+ 19 adilhanney  staff   608 23 Aug 14:22 ..
-rw-r--r--@  1 adilhanney  staff  1196 23 Aug 14:22 Distribution
	com.apple.lastuseddate#PS	  16 
drwx------   5 adilhanney  staff   160 23 Aug 14:22 com.adilhanney.ricochlime.pkg
```
Trying to expand `com.adilhanney.ricochlime.pkg` yields `Could not open package for expansion: com.adilhanney.ricochlime.pkg`

**Comment by huycozy** [comment]
@adil192 Thanks for your updated information. Even though I still can publish via Xcode normally, I can see the different permissions on those files you found above inside the built app (FlutterMacOS.framework) between Flutter versions.

### Check on Flutter versions

- On 3.13.0 / 3.13.1 / master channels: There are sufficient permissions for owner but no permission for group and others.

<details>
<summary>Files permission</summary>

```console
➜  codemagic_cli_distribution.app git:(main) ✗ pwd
/Users/huynq/Desktop/codemagic_cli_distribution/build/macos/Build/Products/Release/codemagic_cli_distribution.app

➜  codemagic_cli_distribution.app git:(main) ✗ ls -al@ ./Contents/Frameworks/FlutterMacOS.framework/Versions/A/
\total 52032
drwx------@ 5 huynq  staff       160 Aug 24 10:43 .
        com.apple.provenance          11 
drwx------@ 4 huynq  staff       128 Aug 23 05:09 ..
        com.apple.provenance          11 
-rwx------@ 1 huynq  staff  26640048 Aug 24 10:43 FlutterMacOS
        com.apple.provenance          11 
drwx------@ 4 huynq  staff       128 Aug 23 05:04 Resources
        com.apple.provenance          11 
drwx------@ 3 huynq  staff        96 Aug 23 05:11 _CodeSignature
        com.apple.provenance          11 

➜  codemagic_cli_distribution.app git:(main) ✗ ls -al@ ./Contents/Frameworks/FlutterMacOS.framework/Versions/A/Resources/
total 1608
drwx------@ 4 huynq  staff     128 Aug 24 08:21 .
        com.apple.provenance        11 
drwx------@ 5 huynq  staff     160 Aug 24 10:51 ..
        com.apple.provenance        11 
-rw-------@ 1 huynq  staff     717 Aug 24 08:21 Info.plist
        com.apple.provenance        11 
-rw-------@ 1 huynq  staff  817232 Aug 24 08:21 icudtl.dat
        com.apple.provenance        11 
```
</details>


- On 3.10.6: Having sufficient permissions for owner, group and others.

<details>
<summary>Files permission</summary>

```console
➜  codemagic_cli_distribution.app git:(main) ✗ pwd
/Users/huynq/Desktop/codemagic_cli_distribution/build/macos/Build/Products/Release/codemagic_cli_distribution.app

➜  codemagic_cli_distribution.app git:(main) ✗ ls -al@ ./Contents/Frameworks/FlutterMacOS.framework/Versions/A/
total 53576
drwxr-xr-x@ 5 huynq  staff       160 Aug 23 22:52 .
        com.apple.provenance          11 
drwxr-xr-x@ 4 huynq  staff       128 Jul 13 02:28 ..
        com.apple.provenance          11 
-rwxr-xr-x@ 1 huynq  staff  27427168 Aug 23 22:52 FlutterMacOS
        com.apple.provenance          11 
drwxr-xr-x@ 4 huynq  staff       128 Jul 13 02:27 Resources
        com.apple.provenance          11 
drwxr-xr-x@ 3 huynq  staff        96 Jul 13 03:57 _CodeSignature
        com.apple.provenance          11 

➜  codemagic_cli_distribution.app git:(main) ✗ ls -al@ ./Contents/Frameworks/FlutterMacOS.framework/Versions/A/Resources/
total 1608
drwxr-xr-x@ 4 huynq  staff     128 Jul 13 02:27 .
        com.apple.provenance        11 
drwxr-xr-x@ 5 huynq  staff     160 Aug 23 22:52 ..
        com.apple.provenance        11 
-rw-r--r--@ 1 huynq  staff     717 Jul  8 05:00 Info.plist
        com.apple.provenance        11 
-rw-r--r--@ 1 huynq  staff  817168 Jul 13 02:07 icudtl.dat
        com.apple.provenance        11 
```
</details>

It seems good with files in `App.framework` as they have sufficient permissions (read, execute). 


<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.13.1, on macOS 13.5 22G74 darwin-x64, locale en-VN)
    • Flutter version 3.13.1 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e1e47221e8 (23 hours ago), 2023-08-22 21:43:18 -0700
    • Engine revision b20183e040
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (3 available)
    • iPhone (mobile) • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.2 19H218
    • macOS (desktop) • macos                                    • darwin-x64     • macOS 13.5 22G74 darwin-x64
    • Chrome (web)    • chrome                                   • web-javascript • Google Chrome 116.0.5845.96

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.14.0-9.0.pre.69, on macOS 13.5 22G74 darwin-x64, locale en-VN)
    • Flutter version 3.14.0-9.0.pre.69 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b32cd68f31 (64 minutes ago), 2023-08-23 22:24:03 -0400
    • Engine revision aa98a9d2e8
    • Dart version 3.2.0 (build 3.2.0-97.0.dev)
    • DevTools version 2.26.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (3 available)
    • iPhone (mobile) • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.2 19H218
    • macOS (desktop) • macos                                    • darwin-x64     • macOS 13.5 22G74 darwin-x64
    • Chrome (web)    • chrome                                   • web-javascript • Google Chrome 116.0.5845.96

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

---

I wonder if this difference is the main cause of the original issue above as I still can publish app to App Store Connect normally. Since [JulienDev](https://github.com/JulienDev) mentioned this worked on previous release 3.10.6, I will mark this as a regression for more information from others.

**Comment by gspencergoog** [comment]
Could this be that your umask is set to something other than 0022?  If you had a umask 0077, would cause what you are seeing, I think.

(Desktop Triage)

**Comment by adil192** [comment]
> Could this be that your umask is set to something other than 0022? If you had a umask 0077, would cause what you are seeing, I think.
> 
> (Desktop Triage)

On my local machine:
```console
Adils-MacBook-Air:~ adilhanney$ umask
0022
Adils-MacBook-Air:~ adilhanney$ umask -S
u=rwx,g=rx,o=rx
```
And on the GitHub Actions runner, I get the same:
```console
0022
u=rwx,g=rx,o=rx
```

**Comment by adil192** [comment]
To add to this, I've made a code sample project on the `empty-macos` branch here: https://github.com/adil192/ricochlime/tree/empty-macos.

I deleted everything except for the `.git` folder (for obvious reasons) and the `.github` folder (which holds the macos build workflow). Then I ran `flutter create . --platforms=macos`, made some minor changes to make the project buildable, and experienced the same "only readable by the root user" error.

![Commits on the empty-macos branch](https://github.com/flutter/flutter/assets/21128619/95f3fea7-8b24-4ca0-839b-ad0bf331956f)

**Comment by adil192** [comment]
Just as an update, I now run `sudo chown -R $USER . && sudo chmod -R a+rwx .` before building, after building, and after packaging.
There is probably a better way to do this, but it successfully published to the App Store.
Since this is already fixed in master, hopefully this won't be necessary much longer

**Comment by huycozy** [comment]
Thanks for your confirmation. Since it's fixed, I will close the issue from here.

**Comment by adil192** [comment]
@huycozy Is there no chance of a hotfix?

**Comment by huycozy** [comment]
You can read about it here: https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#will-a-particular-bug-fix-be-provided-in-a-hotfix-release (and whole that page for more info)

**Comment by nilsreichardt** [comment]
Does this always occur when publishing a macOS app with Flutter? If no, what is required to get this issue? We see the same issue with our app [Sharezone](https://github.com/SharezoneApp/sharezone-app).

I would like to create a cherry-pick request for this issue and need additional information. Also, what exactly commit fixed the issue?

**Comment by adil192** [comment]
Yes this is occuring with all my apps as well as a blank app created with `flutter create`. See the aforementioned `empty-macos` branch here: https://github.com/adil192/ricochlime/tree/empty-macos.

I'm unsure of which commit fixed this as I just tried Flutter 3.10.7 stable, 3.13.1 stable, and master (https://github.com/flutter/flutter/commit/229b74d987720a0437039b6e44e13f20fb94ad51).

**Comment by priitlatt** [comment]
Hi all! Being the author of `app-store-connect publish`, this issue ended up in my hands, and I've been doing some digging on it. Having no Flutter background at all, please excuse me if some of the details or Flutter terminology in the following are a little off.

Having first hand knowledge that `app-store-connect publish` does not do any kind of package modification, the hints from this thread suggested that we should check the contents, or more specifically the file permissions, of build artifacts in `/build/maacos` after `flutter build macos --release` was successfully launched and completed.

First, to reproduce the issue, I used [this repo](https://github.com/huycozy/Codemagic-CLI-Distribution) by @huycozy.
MacOS app was first built with Flutter version `3.10.1` and then with `3.13.2`.
Package from `3.10.1` build was successfully uploaded to App Store, while package from `3.13.2` resulted in the same `The installer package includes files that are only readable by the root user` error as mentioned in [OP](https://github.com/flutter/flutter/issues/132725#issue-1854157498).

Now, given that I'm in a possession of a confirmed "good" and "bad" build I was able to make the following observation:
![Screenshot 2023-09-06 at 12 26 30](https://github.com/flutter/flutter/assets/2756611/1e841719-ba2d-4a85-9142-782948222c55)

While the contents of `build/macos/Build/Products/Release/FlutterMacOS.framework` was readable for all when built with Flutter 3.10.1, it was only accessible for current user when built with Flutter 3.13.2.

I noticed that those files possibly originate from `"${FLUTTER_HOME}"/bin/cache/artifacts/engine/darwin*/FlutterMacOS.framework`. And turns out that the cache archives there indeed are only accessible by (current unix) user with Flutter version `3.12.0+` while both (unix) group and other have read and directory access permissions on Flutter versions up to `3.10.7`.

I altered permissions for files and directories in `"${FLUTTER_HOME}"/bin/cache/artifacts/engine/darwin*/FlutterMacOS.framework` to match how they were in version `3.10.1` using
```shell
chmod -R a+rX "${FLUTTER_HOME}"/bin/cache/artifacts/engine/darwin*/FlutterMacOS.framework
```
This granted read permission to all files and directories, and execute (enter) permission to directories.

After having done so, the build using Flutter `3.13.2` was packaged without problems, and App Store submission also worked.

@TahaTesser pointed out that that the problematic files are hosted in `flutter_infra` Google Cloud bucket , accessible as 
```
https://storage.googleapis.com/flutter_infra_release/flutter/<flutter-engine-version>/darwin-x64-release/FlutterMacOS.framework.zip
```
where Flutter Engine version corresponding to any given Flutter version can be obtained from
```
https://github.com/flutter/flutter/blob/<flutter-version-tag-name>/bin/internal/engine.version
```
for example https://github.com/flutter/flutter/blob/3.13.2/bin/internal/engine.version.


<details>
<summary>Knowing this, we can check the sources using a little script.</summary>

```shell
#!/usr/bin/env sh

download_flutter_macos_framework() {
    VERSION=$1
    ENGINE_VERSION=$(curl -s "https://raw.githubusercontent.com/flutter/flutter/${VERSION}/bin/internal/engine.version")
    echo "Flutter ${VERSION} is using engine version ${ENGINE_VERSION}"
    mkdir "flutter-${VERSION}"
    wget -q "https://storage.googleapis.com/flutter_infra_release/flutter/${ENGINE_VERSION}/darwin-x64-release/FlutterMacOS.framework.zip" -O "flutter-${VERSION}/FlutterMacOS.framework.original.zip"
    md5 "flutter-${VERSION}/FlutterMacOS.framework.original.zip"
    unzip -q "flutter-${VERSION}/FlutterMacOS.framework.original.zip" -d "flutter-${VERSION}"
    md5 "flutter-${VERSION}/FlutterMacOS.framework.zip"
    unzip -q "flutter-${VERSION}/FlutterMacOS.framework.zip" -d "flutter-${VERSION}"
    ls -l "flutter-${VERSION}/Versions/A/FlutterMacOS"
    ls -l "flutter-${VERSION}/Versions/A/Resources/Info.plist"
    echo
}

FLUTTER_INFRA_ARTIFACTS_DIR=$(mktemp -d -t flutter-infra-artifacts)
pushd "${FLUTTER_INFRA_ARTIFACTS_DIR}"

for V in 3.10.5 3.10.6 3.10.7 3.12.0 3.13.0 3.13.1 3.13.2
do
    download_flutter_macos_framework "${V}"
done

popd
rm -r "${FLUTTER_INFRA_ARTIFACTS_DIR}"
```

</details>

Running this script reveals that the artifacts distributed for Flutter versions up to `3.10.7` (included) have correct file permissions for `FlutterMacOS.framework`, and starting from Flutter version `3.12.0` (unix) group and others are lacking all access to `FlutterMacOS.framework`.

<details>
<summary>Full script output</summary>

```shell
Flutter 3.10.5 is using engine version 45f6e009110df4f34ec2cf99f63cf73b71b7a420
MD5 (flutter-3.10.5/FlutterMacOS.framework.original.zip) = 37ea24361f6b5f06cda0a955a682637a
MD5 (flutter-3.10.5/FlutterMacOS.framework.zip) = 2ff609dc0ef5624432f762d621e3474a
-rwxr-xr-x  1 priit  staff  27410960 Jun 14 01:21 flutter-3.10.5/Versions/A/FlutterMacOS
-rw-r--r--  1 priit  staff  717 May 10 03:43 flutter-3.10.5/Versions/A/Resources/Info.plist

Flutter 3.10.6 is using engine version cdbeda788a293fa29665dc3fa3d6e63bd221cb0d
MD5 (flutter-3.10.6/FlutterMacOS.framework.original.zip) = 183573fc56204365dbd637a402ee0a07
MD5 (flutter-3.10.6/FlutterMacOS.framework.zip) = 903dc6917529fcd1a95536a85c76602b
-rwxr-xr-x  1 priit  staff  27410960 Aug 17 20:47 flutter-3.10.6/Versions/A/FlutterMacOS
-rw-r--r--  1 priit  staff  717 Jul  8 01:00 flutter-3.10.6/Versions/A/Resources/Info.plist

Flutter 3.10.7 is using engine version 077a732ef464f5b15de264a5e66ab4e18b3c3cd8
MD5 (flutter-3.10.7/FlutterMacOS.framework.original.zip) = 7d7f8f492250e3a2047bce01824799b1
MD5 (flutter-3.10.7/FlutterMacOS.framework.zip) = 62560bfa0b006bf8bac7ca7f5d980b8d
-rwxr-xr-x  1 priit  staff  27410960 Aug 18 07:26 flutter-3.10.7/Versions/A/FlutterMacOS
-rw-r--r--  1 priit  staff  717 Aug 18 01:32 flutter-3.10.7/Versions/A/Resources/Info.plist

Flutter 3.12.0 is using engine version b84ad1ad30cca8d303b4bcb2bbd77573e4d66122
MD5 (flutter-3.12.0/FlutterMacOS.framework.original.zip) = ca0bdc7a966111d6f86e453de5099f8d
MD5 (flutter-3.12.0/FlutterMacOS.framework.zip) = dc1e03fbc31dc8a1665839fe94460a0e
-rwx------  1 priit  staff  26622736 Jun  8 20:34 flutter-3.12.0/Versions/A/FlutterMacOS
-rw-------  1 priit  staff  717 Jun  8 20:33 flutter-3.12.0/Versions/A/Resources/Info.plist

Flutter 3.13.0 is using engine version 1ac611c64eadbd93c5f5aba5494b8fc3b35ee952
MD5 (flutter-3.13.0/FlutterMacOS.framework.original.zip) = 98751224ee0eeed7b16be58bca084560
MD5 (flutter-3.13.0/FlutterMacOS.framework.zip) = e374ad9e7b191a76d17cfb4a59255d5c
-rwx------  1 priit  staff  26640240 Aug 16 04:16 flutter-3.13.0/Versions/A/FlutterMacOS
-rw-------  1 priit  staff  717 Aug 16 04:09 flutter-3.13.0/Versions/A/Resources/Info.plist

Flutter 3.13.1 is using engine version b20183e04096094bcc37d9cde2a4b96f5cc684cf
MD5 (flutter-3.13.1/FlutterMacOS.framework.original.zip) = 86a51681de8a6e55db70bec35f8eec2e
MD5 (flutter-3.13.1/FlutterMacOS.framework.zip) = 2890a83c566d36ceb5ed786692b7bed2
-rwx------  1 priit  staff  26640224 Aug 23 01:11 flutter-3.13.1/Versions/A/FlutterMacOS
-rw-------  1 priit  staff  717 Aug 23 01:04 flutter-3.13.1/Versions/A/Resources/Info.plist

Flutter 3.13.2 is using engine version b20183e04096094bcc37d9cde2a4b96f5cc684cf
MD5 (flutter-3.13.2/FlutterMacOS.framework.original.zip) = 86a51681de8a6e55db70bec35f8eec2e
MD5 (flutter-3.13.2/FlutterMacOS.framework.zip) = 2890a83c566d36ceb5ed786692b7bed2
-rwx------  1 priit  staff  26640224 Aug 23 01:11 flutter-3.13.2/Versions/A/FlutterMacOS
-rw-------  1 priit  staff  717 Aug 23 01:04 flutter-3.13.2/Versions/A/Resources/Info.plist
```

</details>

**TL;DR: File permissions for `"${FLUTTER_HOME}"/bin/cache/artifacts/engine/darwin*/FlutterMacOS.framework` are messed up ever since (production) version `3.12.0`.**

**Comment by gspencergoog** [comment]
Ahh, if this isn't fixed on master like we thought, we'll take a look and see where it went wrong.  Seems like a packaging issue in the engine.

@priitlatt Thanks for the detailed detective work!

**Comment by nilsreichardt** [comment]
@huycozy Have you also tested the script with the latest master version? You mentioned `3.14.0-0.2.pre` and `3.14.0-14.0.pre`. However, these are the latest beta versions and not master versions. If we trust this comment (https://github.com/flutter/flutter/issues/132725#issuecomment-1694484301) is the issue on master fixed.

@gspencergoog The mentioned comment above says that is fixed on master. I want to create a cherry-pick request for a new stable version because every macOS distribution with 3.13 is affected by this issue. However, we don't know which commit fixed the issue. Do you know which issue fixed this issue or do you have a smaller range where this commit could be? I can't try out 200 commits and create for every commit a new release (even with binary search it takes too long).

**Comment by huycozy** [comment]
@nilsreichardt The reason I used `3.14.0-14.0.pre` since it's the latest tag [here](https://github.com/flutter/flutter/tags) which is released yesterday. And I've seen the [commit](https://github.com/flutter/flutter/commit/229b74d987720a0437039b6e44e13f20fb94ad51) mentioned at https://github.com/flutter/flutter/issues/132725#issuecomment-1694484301 is included in [3.14.0-14.0.pre](https://github.com/flutter/flutter/releases/tag/3.14.0-14.0.pre). If you tried with the latest revision like `3.14.0-14.0.pre.120`, the script will not work. 

I reopened the issue because I think it's not fixed on new releases as I mentioned [above](https://github.com/flutter/flutter/issues/132725#issuecomment-1708595955).

**Comment by adil192** [comment]
> And this issue was probably misunderstood as being fixed on master. @adil192: I think maybe you altered permissions before and were on master at that time (just a guess).

You can see the workflow here: https://github.com/adil192/ricochlime/blob/f85ab1ecb27cc9a1f28d553bf2faef71c3053d21/.github/workflows/macos.yml. There's no `chmod`, `chown`, or `xattr` commands being run.
The only difference in this commit (https://github.com/adil192/ricochlime/commit/f85ab1ecb27cc9a1f28d553bf2faef71c3053d21) was setting the channel to master and disabling `actions/cache`.

**Comment by gspencergoog** [comment]
Okay, I've narrowed the culprit to 3d2504951c0bc794f186da29d35eabec0343e62d, which is an engine roll that includes these two changes:

- https://github.com/flutter/engine/pull/41540
- https://github.com/flutter/engine/pull/41531

Based on the descriptions, my money is on the second one.

Which all means that this issue is possibly a duplicate of https://github.com/flutter/flutter/issues/126705

I can confirm that this issue still happens on the main (master) channel: we're creating macOS projects with unreadable (og-r) file permissions.

<details>
<summary>Detection script</summary>

```bash
#!/usr/bin/env bash

set -e

flutter --version
rev=$(flutter --machine --version | grep frameworkRevision | awk '{print $2}')
rev="rev_${rev:1:8}"
echo "Creating project $rev"
mkdir -p /tmp/bisect > /dev/null 2>&1
cd /tmp/bisect
flutter create "$rev"
cd "$rev"
flutter build macos
cd build
if [[ -z "$(find . -type f ! -perm -g+r -name "Info.plist")" ]]; then
  echo "Found regular permissions"
  exit 0 
else
  echo "Found Info.plist files without valid permissions"
  exit 1
fi
```

</details>

cc @christopherfujino @godofredoc

**Comment by godofredoc** [comment]
https://github.com/flutter/engine/pull/41531 switched to use artifacts build by gn+ninja rather than preparing the package in recipes. If permissions is the only issue I can send a fix in a few hours.

**Comment by godofredoc** [comment]
@priitlatt can you please take a look at this file gs://flutter_archives_v2/flutter_infra_release/flutter/6aedd41942853ba598e2beeab746874e0f24f329/darwin-x64/FlutterMacOS.framework.zip to ensure we are not missing permissions somewhere else?

**Comment by priitlatt** [comment]
@godofredoc Symlink Target of symlink `FlutterMacOS -> Versions/Current/FlutterMacOS` is lacking permissions for group and others. I don't know if that could cause any problems or not. 
```shell
$ ls -l Versions/Current/FlutterMacOS
-rwx------@ 1 priit  staff  103969552 Sep  9 03:37 Versions/Current/FlutterMacOS
```

Also files `Versions/A/**/*` are now executable, which seems a little excessive. All the differences can be seen below:
![Screenshot 2023-09-11 at 10 33 53](https://github.com/flutter/flutter/assets/2756611/268c0466-197d-404d-b7fa-e4aaea322090)

**Comment by godofredoc** [comment]
@priitlatt here is a second try gs://flutter_archives_v2/flutter_infra_release/flutter/fa26d32049be3f58d074011e6586ee36e9bce8e5/darwin-x64/FlutterMacOS.framework.zip please take a look.

**Comment by priitlatt** [comment]
@godofredoc, all file permissions look correct to me in `gs://flutter_archives_v2/flutter_infra_release/flutter/cc02b6b7a29c30ef60d3d01770e5fa6d0e3eca09/darwin-x64/FlutterMacOS.framework.zip`.

**Comment by iampopal** [comment]
We are also having this issue with using code magic, any solution?

**Comment by nilsreichardt** [comment]
@iampopal This is the current workaround: https://github.com/flutter/flutter/issues/132725#issuecomment-1696601540, example: https://github.com/SharezoneApp/sharezone-app/pull/866

**Comment by iampopal** [comment]
Thank you for this fix,
But we still didn't know how to implement this into codemagic. We really
need a fix for this.

**Comment by nilsreichardt** [comment]
@godofredoc Fixed https://github.com/flutter/engine/pull/45598 this issue?

**Comment by godofredoc** [comment]
@nilsreichardt, yes but it still needs to roll to the framework.

@iampopal which channel of the flutter are you using? master, beta, stable?

**Comment by nilsreichardt** [comment]
@godofredoc When the fix is inside the framework, I can use the commit that inlcuded https://github.com/flutter/engine/pull/45598 into the framework for a [cherry-pick request](https://github.com/flutter/flutter/wiki/Flutter-Cherrypick-Process), right?

**Comment by iampopal** [comment]
I am using stable version of flutter, which version shall i use to get this
fix?

**Comment by mubin986** [comment]
This is the issue I'm facing in Codemagic

![Screenshot 2023-09-14 at 10 06 03 PM](https://github.com/flutter/flutter/assets/11482073/f8f20b95-84f9-4ce5-9800-f6a3a9eba5c7)

**Comment by godofredoc** [comment]
Master channel will include the fix soon. I'll report back the flutter/flutter commit that includes the fix.

Stable|Beta channels will get the fix on next week's hotfix releases.

**Comment by iampopal** [comment]
Thank,
Please share update here when this fix is included in upcoming stable
release.

**Comment by nilsreichardt** [comment]
The fix has been landed in the framework (commit: d2ff24172b66635cbe520c80480840bd29f728a8, PR: https://github.com/flutter/flutter/pull/134749). I think we can close this issue 👍

**Comment by godofredoc** [comment]
Thanks @nilsreichardt once post-submit testing has completed I'll close this issue and share the cherry-pick requests to stable and beta.

**Comment by nilsreichardt** [comment]
Cool, thanks for creating a cherry-pick request, @godofredoc 🙏

**Comment by nilsreichardt** [comment]
@Jasguerrero Thanks for releasing a new hotfix (v3.13.5):

> We are pleased to announce the release of Flutter 3.13.5 to the stable channel; This hotfix release addresses the following issues:
> 
> * flutter/134825 - Fixes an issue where apps built in profile mode would not install or run on physical iOS 17 devices.
> 
> In order to get the latest, run ‘flutter upgrade’.

Could it be that you accidentally forgot to add this issue to the changelog? When I'm seeing it correct, have you included in your [cherrypicks-flutter-3.13-candidate.0](https://github.com/Jasguerrero/flutter/commits/cherrypicks-flutter-3.13-candidate.0/) branch a roll engine commit (f433f564a8d44dc56d9bd5d5468659dff1efcbb6) that includes the cherry-pick for this issue: https://github.com/flutter/engine/commit/bd986c5ed20a62dc34b7718c50abc782beae4c33. Just wanted to make sure if I understand correctly that this issue is fixed with the new hotfix.

**Comment by Jasguerrero** [comment]
> @Jasguerrero Thanks for releasing a new hotfix (v3.13.5):
> 
> > We are pleased to announce the release of Flutter 3.13.5 to the stable channel; This hotfix release addresses the following issues:
> > 
> > * flutter/134825 - Fixes an issue where apps built in profile mode would not install or run on physical iOS 17 devices.
> > 
> > In order to get the latest, run ‘flutter upgrade’.
> 
> Could it be that you accidentally forgot to add this issue to the changelog? When I'm seeing it correct, have you included in your [cherrypicks-flutter-3.13-candidate.0](https://github.com/Jasguerrero/flutter/commits/cherrypicks-flutter-3.13-candidate.0/) branch a roll engine commit ([f433f56](https://github.com/flutter/flutter/commit/f433f564a8d44dc56d9bd5d5468659dff1efcbb6)) that includes the cherry-pick for this issue: [flutter/engine@bd986c5](https://github.com/flutter/engine/commit/bd986c5ed20a62dc34b7718c50abc782beae4c33). Just wanted to make sure if I understand correctly that this issue is fixed with the new hotfix.

You are right thank you for pointing this out! I will update the release notes on the [wiki](https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#3135-september-20-2023)
